### DISCUSSION TITLE
Aqara TVOC does not publish changes

### DISCUSSION THREAD

**Discussion by xoxys** [Koenkk/zigbee2mqtt#9171] [Devices]
Hi,

I have two Aqara TVOC air quality monitor paired with zigbee2mqtt. The pairing was working flawlessly and values were published as expected. But after  few days the values are not getting updates anymore. Any idea how to debug it further, or what might be the reason?

Thanks in advance

**Comment by SchoutenHans** [comment]
Hi,  
I have the same problem, two Aqara TVOC's.  One stops sending data after about 2 weeks. The second one after about 4 weeks.
The only way to get new data is to remove and restore batteries. I think it's a Aqara firmware problem...

**Reply by xoxys** [reply]
Not even removing the battery works for me 🙁

**Comment by JDrewes** [comment]
I seem to have a similar or identical issue. I have 3 Aqara TVOC devices (VOCKQJK11LM). At first, all three worked fine for about a week. Then, no more updates were reported for one device through Z2M. 
re-pairing that device worked - but only at the cost of one of the other "falling of the network". 
I have done multiple repairings (sometimes at routers, sometimes at the coordinator), and I cannot get all three devices to keep updating at the same time. 

At the moment it seems, however, that running only two of them does work (I will update if one drops out again).
As there seems to be some effect on the other two when re-pairing one device, I am not sure if the issue is with the TVOCS or maybe somewhere in Z2M? This also because sometimes, the re-pairing didn't work until I restarted Z2M.

I have a total of 8-9 devices connected, all Aqara (except for the coordinator, which is a CC2531 Sonoff-type USB stick). None of them have ever "gone off the network" or stopped reporting, only the TVOCS do that.

Anyone have any tips or clues what this might be? Is it possible that the devices are somehow overwriting each other, or interfering in some way that happens more often when more devices are connected?

Thanks for reading and thanks for Z2M in the first place!

**Comment by barcelona92** [comment]
I thin other problem The coordinator is Slaesh's CC2652RB stick firmware is in the last version

![TVOC-1](https://user-images.githubusercontent.com/33433738/138605204-10092f7b-2015-47f7-b5a0-cf82dde222bf.jpg)
![TVOC-2](https://user-images.githubusercontent.com/33433738/138605213-9796895d-106c-4715-b215-8605170144c5.jpg)

**Comment by danieledwardgeorgehitchcock** [comment]
Just found this thread - I created the original device converter for these, and can confirm that I too, am experiencing the same issue.

I have found that if I click on reconfigure, then wake the device using the physical button, that is enough to get the device back on network however, this isn't practical from a reliability perspective.

I am looking into it, but I fear that there is indeed a hardware issue at play here, either at the devices end (I have read a lot recently about Aqara transmitting specific codes to their devices), or at the co-ordinator level.

It would be great to get these devices reliably speaking to the Z2M network as I plan on controlling extractor fans with the readings.

**Reply by SchoutenHans** [reply]
I hope you find something. I can't get data from one of my devices anymore. Reconfigure and pressing the physical botton don't work.

Logging:

Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"left_network","meta":{"friendly_name":"0x54ef441000105f15"},"type":"device_removed"}'
Info Zigbee: allowing new devices to join.
Info MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"fqpz4-7"}'
Info Device 'AirQuility' joined
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"AirQuility","ieee_address":"0x54ef441000105f15"},"type":"device_joined"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"AirQuility"},"type":"device_connected"}'
Info MQTT publish: topic 'zigbee2mqtt/AirQuility', payload '{"battery":null,"humidity":null,"last_seen":1635857730744,"linkquality":84,"temperature":null,"voc":null}'
Info MQTT publish: topic 'zigbee2mqtt/AirQuility', payload '{"battery":null,"humidity":null,"last_seen":1635857730878,"linkquality":84,"temperature":null,"voc":null}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"AirQuility","ieee_address":"0x54ef441000105f15"},"type":"device_announce"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"AirQuility"},"type":"device_announced"}'
Info MQTT publish: topic 'zigbee2mqtt/AirQuility', payload '{"battery":null,"humidity":null,"last_seen":1635857759922,"linkquality":84,"temperature":null,"voc":null}'
Info MQTT publish: topic 'zigbee2mqtt/AirQuility', payload '{"battery":null,"humidity":null,"last_seen":1635857759926,"linkquality":45,"temperature":null,"voc":null}'
Info MQTT publish: topic 'zigbee2mqtt/AirQuility', payload '{"battery":null,"humidity":null,"last_seen":1635857759931,"linkquality":45,"temperature":null,"voc":null}'
Info MQTT publish: topic 'zigbee2mqtt/AirQuility', payload '{"battery":null,"humidity":null,"last_seen":1635857759935,"linkquality":45,"temperature":null,"voc":null}'
Info 

Device is on MAP, Linkquility is updated. Data if not updated.

**Reply by robertwebbe** [reply]
I have a few of these devices in my house and they were working fine for a number of months. But a short while ago, one disconnected and I cannot get it working again. I've reset the sensor and tried several other things like new pairing, new batteries, closer by, etc. and no luck. 

Next step was to reset a good working one to see if I got that one working fine. But now this one has the same issue. I guess something is the matter with this device. 

How can we get this fixed?

**Comment by JDrewes** [comment]
I created issue #9454 linking to this discussion.

**Comment by Fafitude** [comment]
Bonjour,
Pour information, j'ai le même problème avec un clé conbee2.

**Comment by Reynders1** [comment]
Hi,
The problem still exist with version 1.26.0-1
<img width="450" alt="Screenshot 2022-07-31 at 17 56 27" src="https://user-images.githubusercontent.com/44923513/182034792-e9abdb8b-4a60-4abb-8932-a925af75f615.png">
.

**Reply by xelemorf** [reply]
It seems except in Z2M 1.36-1.38 the issue is still present even in Z2M 2.0. You can only detect this if you add a new device, existing devices will work fine (at least for me), came up when one of these TVOC sensors died and replaced; it was a bad move to go again with Aqara, I sould have known. I will burn this aqara sensor in a fire pit.
